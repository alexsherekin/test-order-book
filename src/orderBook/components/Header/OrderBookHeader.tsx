import React, { FC } from "react";
import styles from "./OrderBookHeader.module.css";

export interface OrderBookHeaderProps {
  groups: number[];
  currentGroup: number;
  groupChanged: (group: number) => void;
};

export const OrderBookHeader: FC<OrderBookHeaderProps> = (props) => {
  return (
    <div className={styles.header}>
      <span>Order book</span>
      <span>{props.children}</span>
      <select
        data-testid="order-book-select"
        value={props.currentGroup}
        onChange={e => props.groupChanged(Number(e.target.value))}>
        {
          props.groups.map((group, index) => <option key={group} value={index}>Group {group}</option>)
        }
      </select>
    </div>
  );
};
