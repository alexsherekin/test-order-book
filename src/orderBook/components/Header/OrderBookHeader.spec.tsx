import { render, fireEvent, getByTestId } from '@testing-library/react';
import React, { ReactPropTypes } from "react";
import { OrderBookHeader, OrderBookHeaderProps } from './OrderBookHeader';

const renderComponent = (props: OrderBookHeaderProps) => {
  return render(<OrderBookHeader {...props} />);
}

const defaultProps: React.PropsWithChildren<OrderBookHeaderProps> = {
  groups: [1,2,3],
  currentGroup: 1,
  groupChanged: jest.fn(),
  children: <span>Children</span>,
}

describe(OrderBookHeader.name, () => {
  it("should call groupChanged callback", () => {
    const props: OrderBookHeaderProps = {
      ...defaultProps
    };
    const selectedValue = 2;
    const renderedComponent = renderComponent(props);

    fireEvent.change(getByTestId(renderedComponent.container, "order-book-select"), {
      target: {
        value: selectedValue,
      }
    });

    expect(props.groupChanged).toHaveBeenCalledWith(selectedValue);
  });

  it("should render", () => {
    const renderedComponent = renderComponent(defaultProps);

    expect(renderedComponent.asFragment()).toMatchSnapshot();
  });
});
