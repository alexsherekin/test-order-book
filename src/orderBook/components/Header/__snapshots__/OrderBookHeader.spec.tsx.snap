// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookHeader should render 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <span>
      Order book
    </span>
    <span>
      <span>
        Children
      </span>
    </span>
    <select
      data-testid="order-book-select"
    >
      <option
        value="0"
      >
        Group 1
      </option>
      <option
        value="1"
      >
        Group 2
      </option>
      <option
        value="2"
      >
        Group 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
